grin/src/Pipeline/Definitions.hs

```diff
@@ -0,0 +1,180 @@
+{-# LANGUAGE TemplateHaskell, PatternSynonyms #-}
+module Pipeline.Definitions where
```

Comment:
I am against this change. Decoupling definitions from the only one module which uses them considered a bad practice in Haskell.

Reply:
The Pipeline module is already way too big. Extracting the definitions and utilities out not only reduces the size of the module but also makes it more comprehensible. The data and the functionality are separated this way. Furthermore, the definitions are so simple, they can be understood by reading them only once. Every other occasion the programmer has to scroll through those lines is unnecessary overhead (also personal experience).